### PR TITLE
Use examples without types in the post

### PR DESCRIPTION
**Pull request by gaearon**
In response to feedback in https://news.ycombinator.com/item?id=16693633.

Seeing type signatures upfront is a little intimidating, especially because it gives the feeling you need to know what `$Shape` means etc. I hid it in a gist.

### CODE REVIEW

**Comment by gaearon** [comment]
screen shot 2018-03-28 at 00 35 24
screen shot 2018-03-28 at 00 35 29

File: examples/update-on-async-rendering/definition-getsnapshotbeforeupdate.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Example extends React.Component {
+  getSnapshotBeforeUpdate(prevProps, prevState) {
+    // ...
```
Comment (acdlite) [inline examples/update-on-async-rendering/definition-getsnapshotbeforeupdate.js]: Maybe return snapshot here?

Comment (gaearon) [reply examples/update-on-async-rendering/definition-getsnapshotbeforeupdate.js]: Then it'll be a fully fledged example. Maybe I should just remove `componentDidUpdate`? There's a full example below anyway.

Comment (gaearon) [reply examples/update-on-async-rendering/definition-getsnapshotbeforeupdate.js]: I'll remove cDU to focus this on the new method signature.

**Comment by bvaughn** [comment]
Thanks dan!

**Pull request merged**